Dockerfile.Refind

```diff
@@ -0,0 +1,71 @@
+FROM ubuntu:20.04
+MAINTAINER Dmitrii Okunev <xaionaro@dx.center>
+
+RUN \
+	DEBIAN_FRONTEND=noninteractive apt-get update && \
+	DEBIAN_FRONTEND=noninteractive apt-get install -y \
+	    bison \
+		build-essential \
+		ccache \
+		curl \
+		flex \
+		gcc-7 libgcc-7-dev \
+		gcc-9 libgcc-9-dev gcc-9-aarch64-linux-gnu gcc-9-arm-linux-gnueabihf \
+		gcc-10 libgcc-10-dev gcc-10-aarch64-linux-gnu gcc-10-arm-linux-gnueabihf \
+		gcc cross-gcc-dev gcc-aarch64-linux-gnu gcc-arm-linux-gnueabihf \
+		git \
+		iasl \
+		nasm \
+		sudo \
+		python \
+		python-dev \
+		python3 \
+		python3-distutils \
+		python3-pip \
+		qemu \
+		uuid-dev \
+		vim \
+		wget \
+		zip \
+	&& DEBIAN_FRONTEND=noninteractive apt-get clean
+
+RUN curl https://bootstrap.pypa.io/pip/2.7/get-pip.py --output /tmp/get-pip.py && python /tmp/get-pip.py
+RUN pip3 install -q uefi_firmware && pip install -q uefi_firmware
+
+RUN useradd -m edk2 && \
+	adduser edk2 sudo && \
+	sed -e 's/ALL$/NOPASSWD: ALL/' -i /etc/sudoers
+
+RUN mkdir /home/edk2/.ccache && \
+	chown edk2:edk2 /home/edk2/.ccache
+
+VOLUME /home/edk2/.ccache
+
+USER edk2
+WORKDIR /home/edk2
+
+ENV PYTHON_COMMAND python3
+
+RUN git clone "https://github.com/tianocore/edk2" edk2
+RUN git clone "https://github.com/tianocore/edk2-libc" libc
+RUN git clone "https://github.com/tianocore/edk2-platforms" platforms
+
+ARG EDK2_VERSION=${EDK2_VERSION}
+ARG IMAGE_NAME=${IMAGE_NAME}
+RUN echo "EDK2_VERSION:<$EDK2_VERSION>" && if [ "$EDK2_VERSION" != '' -a "$EDK2_VERSION" != 'latest' ]; then git -C edk2 checkout "$EDK2_VERSION"; fi
+RUN git -C edk2 submodule update --init
+ADD build-edk2.sh /home/edk2/build-edk2.sh
+RUN /home/edk2/build-edk2.sh
+
+RUN mkdir -p /home/edk2/src /home/edk2/gcc/7 /home/edk2/gcc/9 /home/edk2/gcc/10 && \
+    ln -s /usr/bin/gcc-7 /home/edk2/gcc/7/gcc && \
+    ln -s /usr/bin/gcc-9 /home/edk2/gcc/9/gcc && \
+    ln -s /usr/bin/gcc-10 /home/edk2/gcc/10/gcc
+
+ENV WORKSPACE "/home/edk2"
+ENV TOOLCHAIN "GCC5"
+ENV TARGET_ARCH "X64"
+ENV BUILD_TARGET "DEBUG"
+ADD build-edk2.sh /home/edk2/build-edk2.sh
+ADD entry.sh /home/edk2/entry.sh
+CMD /bin/bash /home/edk2/entry.sh
```

Comment:
Could you please elaborate why we need a separate `Dockerfile` (why we cannot do what we need in the main one)?

Reply:
I put it only because if I rebuild the rEFInd image with the main Dockerfile it does not compile. There is a regression from the last working image, but I couldn't find what exactly is wrong.

Reply:
CC:@xaionaro So using the main Dockerfile for rebuilding does not work for these examples:
https://github.com/startergo/edk2-builder-docker/tree/main/examples/github.com-andreiw-UefiToolsPkg

https://github.com/startergo/edk2-builder-docker/tree/main/examples/sourceforge.net-srs5694-Refind

Using:
```
cp ../../build-edk2.sh . && cp ../../clone-edk2.sh . && cp ../../entry.sh .
	EDK2_VERSION=vUDK2018 DOCKERFILE_PATH=../../Dockerfile IMAGE_NAME=xaionaro2/edk2-builder:vUDK2018 ../../hooks/build
```

<details>
  <summary>Workflow</summary>

```
In file included from /home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.h:22,
                 from /home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:15:
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c: In function 'BasePrintLibConvertValueToStringS':
/home/edk2/edk2/MdePkg/Include/Library/PcdLib.h:388:45: error: '_PCD_GET_MODE_32_PcdMaximumAsciiStringLength' undeclared (first use in this function)
  388 | #define PcdGet32(TokenName)                 _PCD_GET_MODE_32_##TokenName
      |                                             ^~~~~~~~~~~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:24:32: note: in expansion of macro 'PcdGet32'
   24 | #define ASCII_RSIZE_MAX       (PcdGet32 (PcdMaximumAsciiStringLength))
      |                                ^~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:377:9: note: in expansion of macro 'ASCII_RSIZE_MAX'
  377 |     if (ASCII_RSIZE_MAX != 0) {
      |         ^~~~~~~~~~~~~~~
/home/edk2/edk2/MdePkg/Include/Library/PcdLib.h:388:45: note: each undeclared identifier is reported only once for each function it appears in
  388 | #define PcdGet32(TokenName)                 _PCD_GET_MODE_32_##TokenName
      |                                             ^~~~~~~~~~~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:24:32: note: in expansion of macro 'PcdGet32'
   24 | #define ASCII_RSIZE_MAX       (PcdGet32 (PcdMaximumAsciiStringLength))
      |                                ^~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:377:9: note: in expansion of macro 'ASCII_RSIZE_MAX'
  377 |     if (ASCII_RSIZE_MAX != 0) {
      |         ^~~~~~~~~~~~~~~
/home/edk2/edk2/MdePkg/Include/Library/PcdLib.h:388:45: error: '_PCD_GET_MODE_32_PcdMaximumUnicodeStringLength' undeclared (first use in this function)
  388 | #define PcdGet32(TokenName)                 _PCD_GET_MODE_32_##TokenName
      |                                             ^~~~~~~~~~~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:23:32: note: in expansion of macro 'PcdGet32'
   23 | #define RSIZE_MAX             (PcdGet32 (PcdMaximumUnicodeStringLength))
      |                                ^~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:384:9: note: in expansion of macro 'RSIZE_MAX'
  384 |     if (RSIZE_MAX != 0) {
      |         ^~~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c: In function 'BasePrintLibSPrintMarker':
/home/edk2/edk2/MdePkg/Include/Library/PcdLib.h:388:45: error: '_PCD_GET_MODE_32_PcdMaximumUnicodeStringLength' undeclared (first use in this function)
  388 | #define PcdGet32(TokenName)                 _PCD_GET_MODE_32_##TokenName
      |                                             ^~~~~~~~~~~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:23:32: note: in expansion of macro 'PcdGet32'
   23 | #define RSIZE_MAX             (PcdGet32 (PcdMaximumUnicodeStringLength))
      |                                ^~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:594:9: note: in expansion of macro 'RSIZE_MAX'
  594 |     if (RSIZE_MAX != 0) {
      |         ^~~~~~~~~
/home/edk2/edk2/MdePkg/Include/Library/PcdLib.h:388:45: error: '_PCD_GET_MODE_32_PcdMaximumAsciiStringLength' undeclared (first use in this function)
  388 | #define PcdGet32(TokenName)                 _PCD_GET_MODE_32_##TokenName
      |                                             ^~~~~~~~~~~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:24:32: note: in expansion of macro 'PcdGet32'
   24 | #define ASCII_RSIZE_MAX       (PcdGet32 (PcdMaximumAsciiStringLength))
      |                                ^~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:599:9: note: in expansion of macro 'ASCII_RSIZE_MAX'
  599 |     if (ASCII_RSIZE_MAX != 0) {
      |         ^~~~~~~~~~~~~~~
make: *** [GNUmakefile:309: /home/edk2/Build/UefiToolsPkg/DEBUG_GCC5/X64/MdePkg/Library/BasePrintLib/BasePrintLib/OUTPUT/PrintLibInternal.obj] Error 1


build.py...
 : error 7000: Failed to execute command
	make tbuild [/home/edk2/Build/UefiToolsPkg/DEBUG_GCC5/X64/MdePkg/Library/BasePrintLib/BasePrintLib]


build.py...
 : error 7000: Failed to execute command
	make tbuild [/home/edk2/Build/UefiToolsPkg/DEBUG_GCC5/X64/MdePkg/Library/UefiDevicePathLib/UefiDevicePathLib]


build.py...
 : error 7000: Failed to execute command
	make tbuild [/home/edk2/Build/UefiToolsPkg/DEBUG_GCC5/X64/MdePkg/Library/BaseLib/BaseLib]


build.py...
 : error F002: Failed to build module
	/home/edk2/edk2/MdePkg/Library/BasePrintLib/BasePrintLib.inf [X64, GCC5, DEBUG]

- Failed -
Build end time: 14:06:45, Dec.14 2021
Build total time: 00:00:18

+ exit
make: *** [all] Error 1
```

</details>

This example never worked:

https://github.com/startergo/edk2-builder-docker/tree/main/examples/github.com-joevt-RefindPlus

```
cp ../../build-edk2.sh . && cp ../../clone-edk2.sh . && cp ../../entry.sh .
	EDK2_VERSION=RefindPlusUDKJVT DOCKERFILE_PATH=../../Dockerfile IMAGE_NAME=xaionaro2/edk2-builder:RefindPlusUDKJVT ../../hooks/build

```